languages/cpp/src/BitwardenClient.cpp

```diff
@@ -0,0 +1,127 @@
+#include "BitwardenClient.h"
+#include <iostream>
+#include <string>
+
+BitwardenClient::BitwardenClient(ClientSettings clientSettings) : library(nullptr), commandRunner(nullptr), isClientOpen(false), projects(nullptr), secrets(nullptr) {
+
+    nlohmann::json jsonClientSettings;
+    Bitwarden::Sdk::to_json(jsonClientSettings, clientSettings);
+
+    std::string jsonClientSettingsString = jsonClientSettings.dump();
+    const char* jsonClientSettingsCStr = jsonClientSettingsString.c_str();
+
+    try {
+        library = new BitwardenLibrary("./");
+        client = library->init(jsonClientSettingsCStr);
+        commandRunner = new CommandRunner(library, client);
+        projects = Projects(commandRunner);
+        secrets = Secrets(commandRunner);
+        isClientOpen = true;
+    } catch (const std::exception& ex) {
+        std::cerr << "Failed to initialize: " << ex.what() << std::endl;
+        throw ex;
+    }
+}
+
+BitwardenClient::~BitwardenClient() {
+    if (library) {
+        delete commandRunner;
+        library->free_mem(client);
+        delete library;
+        isClientOpen = false;
+    }
+}
+
+ResponseForApiKeyLoginResponse BitwardenClient::accessTokenLogin(const std::string& accessToken) {
```

Comment:
In other clients we decided that the return type for this function is not very useful, so we should make this return `void` to keep in line with the others

Reply:
As I understood, you want the function `accessTokenLogin ` not to return anything.

Like this:


```
void BitwardenClient::accessTokenLogin(const std::string& accessToken) {
    Command command;
    AccessTokenLoginRequest accessTokenLoginRequest;
    accessTokenLoginRequest.set_access_token(accessToken);
    command.set_access_token_login(accessTokenLoginRequest);

    auto deserializer = [](const char* response) -> ApiKeyLoginResponse {
        nlohmann::json jsonResponse = nlohmann::json::parse(response);
        ApiKeyLoginResponse loginResponse;
        Bitwarden::Sdk::from_json(jsonResponse, loginResponse);
        return loginResponse;
    };
    try {
        commandRunner->runCommand<ApiKeyLoginResponse>(command, deserializer);
    } catch (const std::exception& ex) {
        std::cerr << "Error in accessTokenLogin: " << ex.what() << std::endl;
        throw ex;
    }
}
```

Reply:
Yeah, that looks good!

Reply:
OK, I pushed the changes, please check.

Reply:
Perfect, both changes look good!